gsoc/proposal-Apply Machine Learning to Monitoring website changes.md

```diff
@@ -0,0 +1,35 @@
+# Student
+
+| Title | Description |
+|--- | ---|
+| **Name** | Romit Jain |
+| **Affiliation** | The LNM Institute of Information Technology, Electronics and Communication Engineering, Bachelors’ of Technology |
+| **Location** | Jaipur, India and IST |
+| **GitHub handle** | @romitjain |
+| **Slack handle** | @romitjain |
+| **Email** | romit.73@gmail.com |
+| **Other Contact** | https://www.linkedin.com/in/romitjain7/ |
+
+# Project Details
+
+| Title | Description |
+|--- | ---|
+| **Project Title** | Apply Machine Learning to Monitoring Website Changes |
+| **Project Abstract/Summary** | The project aims at detecting the changes in a website in an automated way and help analysts in monitoring the important changes efficiently. |
+| **Describe the need your project fills** | The project will be helpful in detecting some major changes in government agency website. These changes can help study various important factors. Changes can be big ones (important) or small ones (unimportant). Through applying Machine Learning, one can train a ML model to just send the important changes in a website to analyst rather than sending all the changes. This will be helpful on analyst front and will save a lot of time. Surely, we will be needing to train the model first. That will require analyst help by filtering important changes from unimportant ones. Classifying the type of changes (which is already done), can also serve as a feature. |
+| **Describe how this project meets this need** | One of the most important things about this project is to train the Machine Learning model. This is a classification problem, classifying what is important and what is not. Starting with multiple machine learning algorithms and carefully analyzing how good each one of them is doing (accuracy). At the beginning, voting mechanism can be deployed, to take a vote from multiple ML algorithms to classify the change as meaningful or not meaningful. Kernel functions would be an integral and core part as we would be comparing 2 snapshots of webpages. Text analysis will also be an important part for training the model. Starting with small labelled dataset, once we work out all the issues like variance, overfitting etc., the ML algorithm can be deployed to the larger dataset, to filter the meaningful changes in two versions (snapshots) of webpages with certain accuracy. |
```

Comment:
In broad outline, this is probably in the right direction. However, it is rather generic. It would be helpful to have a little bit more information about your thoughts for handling the unique characteristics of the current data sources.

You may wish to have a look at discussions that have already taken place on this topic over in the [issues list for the web-monitoring-processing](https://github.com/edgi-govdata-archiving/web-monitoring-processing/issues) repository. Look for the "ML" ones.